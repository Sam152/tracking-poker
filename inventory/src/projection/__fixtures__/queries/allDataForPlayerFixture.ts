export const allDataForPlayerFixture = [
    {
        date: "2023/06/15",
        entity_type: "player_appearance",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "appearance#player#sashimi#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        show_name: "Sashimi, Kiran C2B, Israeli Ron & Mike X Play $10/20/40 - Commentary by Christian Soto",
        sk: "appearance#slug#dVs7ORBHsF0#",
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cc#player#sashimi#",
        gsi2pk: "stat_type#cc",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cc#show#sashimi#",
        type: "cc",
        value: 34840,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#sashimi#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#sashimi#",
        type: "cw",
        value: 5840,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#pfr#player#sashimi#",
        gsi2pk: "stat_type#pfr",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#pfr#show#sashimi#",
        type: "pfr",
        value: 20,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#vpip#player#sashimi#",
        gsi2pk: "stat_type#vpip",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#vpip#show#sashimi#",
        type: "vpip",
        value: 36,
    },
];
