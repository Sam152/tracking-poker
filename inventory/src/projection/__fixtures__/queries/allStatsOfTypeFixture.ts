export const allStatsOfTypeFixture = [
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#alex#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#alex#slug#dVs7ORBHsF0#",
        pk: "player#alex",
        player: "alex",
        player_name: "ALEX",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#alex#",
        type: "cw",
        value: 27530,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#c2b#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#c2b#slug#dVs7ORBHsF0#",
        pk: "player#c2b",
        player: "c2b",
        player_name: "C2B",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#c2b#",
        type: "cw",
        value: 11760,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#faizal#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#faizal#slug#dVs7ORBHsF0#",
        pk: "player#faizal",
        player: "faizal",
        player_name: "FAIZAL",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#faizal#",
        type: "cw",
        value: -5180,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#johnny#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#johnny#slug#dVs7ORBHsF0#",
        pk: "player#johnny",
        player: "johnny",
        player_name: "JOHNNY",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#johnny#",
        type: "cw",
        value: 12480,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#mikex#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#mikex#slug#dVs7ORBHsF0#",
        pk: "player#mikex",
        player: "mikex",
        player_name: "MIKE X",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#mikex#",
        type: "cw",
        value: -21840,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#nick#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#nick#slug#dVs7ORBHsF0#",
        pk: "player#nick",
        player: "nick",
        player_name: "NICK",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#nick#",
        type: "cw",
        value: 5540,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#sashimi#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#sashimi#",
        type: "cw",
        value: 5840,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#tal#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#tal#slug#dVs7ORBHsF0#",
        pk: "player#tal",
        player: "tal",
        player_name: "TAL",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#tal#",
        type: "cw",
        value: -13560,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#wei#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#wei#slug#dVs7ORBHsF0#",
        pk: "player#wei",
        player: "wei",
        player_name: "WEI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#wei#",
        type: "cw",
        value: 2290,
    },
];
