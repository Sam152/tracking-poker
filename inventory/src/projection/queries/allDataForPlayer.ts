import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Stat from "../entity/stat";
import { statGroupFromItems } from "../entity/stat";
import * as Player from "../entity/playerAppearance";
import * as PlayerAppearance from "../entity/playerAppearance";

type StatGroup = Partial<Record<Stat.StatType, Stat.Stat[]>>;

export function allDataForPlayer(player: Player.PlayerId): StructuredQuery<{
    appearances: PlayerAppearance.PlayerAppearance[];
    stats: StatGroup;
}> {
    return {
        query: new QueryCommand({
            TableName: "inventory",
            KeyConditionExpression: "pk = :pk",
            ExpressionAttributeValues: {
                ":pk": `player#${player}`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            return {
                appearances: items
                    .filter((item) => item.entity_type === "player_appearance")
                    .map((appearanceStorage) => PlayerAppearance.fromStorage(appearanceStorage)),
                stats: statGroupFromItems(items),
            };
        },
    };
}
