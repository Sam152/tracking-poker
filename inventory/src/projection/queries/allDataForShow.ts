import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Stat from "../entity/stat";
import { statGroupFromItems } from "../entity/stat";
import * as Show from "../entity/show";
import * as PlayerAppearance from "../entity/playerAppearance";

type StatGroup = Partial<Record<Stat.StatType, Stat.Stat[]>>;

export function allDataForShow(showId: Show.ShowId): StructuredQuery<{
    show?: Show.Show;
    players: PlayerAppearance.PlayerAppearance[];
    stats: StatGroup;
}> {
    return {
        query: new QueryCommand({
            TableName: "inventory",
            IndexName: "gsi1",
            KeyConditionExpression: "gsi1pk = :gsi1pk",
            ExpressionAttributeValues: {
                ":gsi1pk": `slug#${showId}`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            const show = items.find((item) => item.entity_type === "show");
            return {
                show: show ? Show.fromStorage(show) : undefined,
                players: items
                    .filter((item) => item.entity_type === "player_appearance")
                    .map((appearanceStorage) => PlayerAppearance.fromStorage(appearanceStorage)),
                stats: statGroupFromItems(items),
            };
        },
    };
}
