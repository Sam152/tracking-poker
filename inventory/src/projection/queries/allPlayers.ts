import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Player from "../entity/player";

export function allPlayers(): StructuredQuery<{
    players: Player.Player[];
}> {
    return {
        query: new QueryCommand({
            TableName: "inventory",
            KeyConditionExpression: "pk = :pk AND begins_with(sk, :skBeginsWith)",
            ExpressionAttributeValues: {
                ":pk": `all_operators`,
                ":skBeginsWith": `player#`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            return {
                players: items
                    .filter((item) => item.entity_type === "player")
                    .map((playerStorage) => Player.fromStorage(playerStorage)),
            };
        },
    };
}
