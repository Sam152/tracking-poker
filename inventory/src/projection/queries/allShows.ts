import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Show from "../entity/show";

export function allShows(): StructuredQuery<Show.Show[]> {
    return {
        query: new QueryCommand({
            // Scan index backwards, to show most recent shows first.
            ScanIndexForward: false,
            TableName: "inventory",
            KeyConditionExpression: "pk = :pk AND begins_with(sk, :skBeginsWith)",
            ExpressionAttributeValues: {
                ":pk": `operator#hcl`,
                ":skBeginsWith": `show#`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            return items.map((storageShow) => Show.fromStorage(storageShow));
        },
    };
}
