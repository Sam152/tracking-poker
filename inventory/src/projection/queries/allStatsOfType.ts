import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Stat from "../entity/stat";

export function allStatsOfType(type: Stat.StatType): StructuredQuery<Stat.Stat[]> {
    return {
        query: new QueryCommand({
            TableName: "inventory",
            IndexName: "gsi2",
            KeyConditionExpression: "gsi2pk = :gsi2pk",
            ExpressionAttributeValues: {
                ":gsi2pk": `stat_type#${type}`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            return items.map((statStorage) => Stat.fromStorage(statStorage));
        },
    };
}
