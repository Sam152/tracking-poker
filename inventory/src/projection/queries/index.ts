import { QueryCommand } from "@aws-sdk/lib-dynamodb";

export type StructuredQuery<T> = {
    query: QueryCommand;
    parser: (items: Record<string, any>[]) => T;
};
