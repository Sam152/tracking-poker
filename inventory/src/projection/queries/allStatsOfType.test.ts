import { allStatsOfType } from "./allStatsOfType";
import { allStatsOfTypeFixture } from "../__fixtures__/queries/allStatsOfTypeFixture";

describe("allStatsOfType", () => {
    test("parser", () => {
        expect(allStatsOfType("cw").parser(allStatsOfTypeFixture)).toMatchSnapshot();
    });
});
