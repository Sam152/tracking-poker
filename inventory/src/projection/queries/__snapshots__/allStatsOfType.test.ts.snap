// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allStatsOfType parser 1`] = `
[
  {
    "player": "alex",
    "player_name": "ALEX",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": 27530,
  },
  {
    "player": "c2b",
    "player_name": "C2B",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": 11760,
  },
  {
    "player": "faizal",
    "player_name": "FAIZAL",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": -5180,
  },
  {
    "player": "johnny",
    "player_name": "JOHNNY",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": 12480,
  },
  {
    "player": "mikex",
    "player_name": "MIKE X",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": -21840,
  },
  {
    "player": "nick",
    "player_name": "NICK",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": 5540,
  },
  {
    "player": "sashimi",
    "player_name": "SASHIMI",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": 5840,
  },
  {
    "player": "tal",
    "player_name": "TAL",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": -13560,
  },
  {
    "player": "wei",
    "player_name": "WEI",
    "show": "dVs7ORBHsF0",
    "type": "cw",
    "value": 2290,
  },
]
`;
